### PR TITLE
fix(direct-upload): use same concurrency as deferred upload

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
PR makes direct upload concurrency level the same as differed uploads.
multiple parallel uploads are limited by the same worker so the node remains protected against goroutine pileup.

### CODE REVIEW

File: pkg/storer/netstore.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 const (
-	directUploadWorkers = 16
```
Comment (aloknerurkar) [inline pkg/storer/netstore.go:18]: This is not the same. We can have multiple calls to DirectUpload. It creates 16 workers for each upload.

**Review by aloknerurkar — CHANGES_REQUESTED** [review]

File: pkg/storer/netstore.go (hunk 1 above)
Comment (istae) [reply pkg/storer/netstore.go:18]: Does 16 workers mean 16 chunk push attempts in parallel?

Comment (aloknerurkar) [reply pkg/storer/netstore.go:18]: Yes 16 push attempts in parallel per upload. So the call to DirectUpload returns a putter which can use upto 16 routines to push in parallel.

Comment (istae) [reply pkg/storer/netstore.go:18]: in that case, we do not actually need this limiter. the pusher is already limited.

**Comment by istae** [comment]
imo it'd would been ok to simply require a batch ID starting with the new release

**Review by aloknerurkar — APPROVED** [review]

**Pull request merged**